validator/lib/validateTileset.js

```diff
@@ -168,3 +191,38 @@ function validateTileHierarchy(root, tilesetDirectory) {
             return message;
         });
 }
+
+function checkBoundingVolume(tileBV, parentBV, tileTransform, parentTransform, type) {
```

Comment:
This could be more generic so that it takes the bounding volumes instead of the boxes. Then the code here would basically include this code from above (plus code for handling the transforms) :

```
            var contentRegion = content.boundingVolume.region;
            var contentSphere = content.boundingVolume.sphere;
            var contentBox = content.boundingVolume.box;
            var tileRegion = tile.boundingVolume.region;
            var tileSphere = tile.boundingVolume.sphere;
            var tileBox = tile.boundingVolume.box;

            if (defined(contentRegion) && defined(tileRegion) && !regionInsideRegion(contentRegion, tileRegion)) {
                return 'content region [' + contentRegion + '] is not within tile region + [' + tileRegion + ']';
            }

            if (defined(contentSphere) && defined(tileSphere) && !sphereInsideSphere(contentSphere, tileSphere)) {
                return 'content sphere [' + contentSphere + '] is not within tile sphere + [' + tileSphere + ']';
            }

            if (defined(contentBox) && defined(tileBox) && !boxInsideBox(contentBox, tileBox)) {
                return 'content box [' + contentBox + '] is not within tile box [' + tileBox + ']';
            }

            if (defined(contentBox) && defined(tileSphere) && !boxInsideSphere(contentBox, tileSphere)) {
                return 'content box [' + contentBox + '] is not within tile sphere [' + tileSphere + ']';
            }
```